kolibri/content/utils/channel_import.py

```diff
@@ -0,0 +1,287 @@
+        # have been set up to handle the fact that this is None.
+        try:
+            SourceRecord = self.source.get_class(model)
+        except ClassNotFoundError:
+            SourceRecord = None
+
+        columns = dest_table.columns.keys()
+        data_to_insert = []
+        merge = model in merge_models
+        for record in table_mapper(SourceRecord):
+            data = {
+                str(column): row_mapper(record, column) for column in columns if row_mapper(record, column) is not None
+            }
+            if merge:
+                # Models that should be merged (see list above) need to be individually merged into the session
+                # as SQL Alchemy ORM does not support INSERT ... ON DUPLICATE KEY UPDATE style queries,
+                # as not available in SQLite, only MySQL as far as I can tell:
+                # http://hackthology.com/how-to-compile-mysqls-on-duplicate-key-update-in-sql-alchemy.html
+                self.destination.session.merge(DestinationRecord(**data))
+            else:
+                data_to_insert.append(data)
+            unflushed_rows += 1
+            if unflushed_rows == 10000:
+                if not merge:
+                    self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
+                    data_to_insert = []
+                self.destination.session.flush()
+                unflushed_rows = 0
+        if not merge and data_to_insert:
+            self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
+        return unflushed_rows
+
+    def import_channel_data(self):
+
+        unflushed_rows = 0
+
+        for model in self.content_models:
+            mapping = self.schema_mapping.get(model, {})
+            row_mapper = self.generate_row_mapper(mapping.get('per_row'))
+            table_mapper = self.generate_table_mapper(mapping.get('per_table'))
+            logging.info('Importing {model} data'.format(model=model.__name__))
+            unflushed_rows = self.table_import(model, row_mapper, table_mapper, unflushed_rows)
+        self.destination.session.commit()
+
+    def delete_content_tree_and_files(self):
+        delete_content_tree_and_files(self.channel_id)
+
+    def end(self):
+        self.source.end()
+        self.destination.end()
+
+
+class NoVersionChannelImport(ChannelImport):
+    """
+    Class defining the schema mapping for importing old content databases (i.e. ones produced before the
+    ChannelImport machinery was implemented). The schema mapping below defines how to bring in information
+    from the old version of the Kolibri content databases into the database for the current version of Kolibri.
+    """
+
+    schema_mapping = {
+        ContentNode: {
+            'per_row': {
+                'channel_id': 'infer_channel_id_from_source',
+                'tree_id': 'get_tree_id',
+                'available': 'none',
```

Comment:
It maps to a method on the import class that then returns None. Didn't want to allow direct assignment of values in here because that would lead to a lot more complication in the mapping logic if these are anything other than keys.